DPPP/MSBDAReader.cc

```diff
@@ -174,52 +174,45 @@ bool MSBDAReader::process(const DPBuffer&) {
   ScalarColumn<int> ant1_col(ms_, MS::columnName(MS::ANTENNA1));
   ScalarColumn<int> ant2_col(ms_, MS::columnName(MS::ANTENNA2));
   ArrayColumn<float> weights_col(ms_, MS::columnName(MS::WEIGHT_SPECTRUM));
+  ArrayColumn<Complex> data_col(ms_, MS::columnName(MS::DATA));
   ArrayColumn<double> uvw_col(ms_, MS::columnName(MS::UVW));
   ScalarColumn<double> time_col(ms_, MS::columnName(MS::TIME));
   ScalarColumn<double> interval_col(ms_, MS::columnName(MS::INTERVAL));
   ScalarColumn<double> exposure_col(ms_, MS::columnName(MS::EXPOSURE));
   ScalarColumn<int> data_desc_id_col(ms_, MS::columnName(MS::DATA_DESC_ID));
 
-  // Cache the data that will be add to the buffer
-  RefRows cell_range{
-      nread_, std::min(ms_.nrow() - 1, rownr_t(nread_ + info().nbaselines()))};
-
-  casacore::Array<casacore::Complex> data;
-  if (read_vis_data_) {
-    ArrayColumn<Complex> data_col(ms_, MS::columnName(MS::DATA));
-    data = data_col.getColumnCells(cell_range);
-  }
-  auto weights = weights_col.getColumnCells(cell_range);
-  auto uvw = uvw_col.getColumnCells(cell_range);
-  auto time = time_col.getColumnCells(cell_range);
-  auto interval = interval_col.getColumnCells(cell_range);
-  auto exposure = exposure_col.getColumnCells(cell_range);
-  auto data_desc_id = data_desc_id_col.getColumnCells(cell_range);
-
-  unsigned i = 0;
-  while (nread_ < ms_.nrow() && i < info().nbaselines()) {
-    const double ms_time = time[i];
+  while (nread_ < ms_.nrow() && buffer->GetRemainingCapacity() > 0) {
```

Comment:
- `buffer->GetRemaininingCapacity() > 0` makes little sense: If the remaining capacity is 1, AddRow will probably not succeed, as a row needs n_channels * n_correlations capacity.
- Removing 'i' as the loop variable is possible, but then you should create some loop end variable. For example:
```
auto read_end = std::min(ms_.nrow(), nread + info().nbaselines());
for (; nread_ < read_end; ++nread_) {
```